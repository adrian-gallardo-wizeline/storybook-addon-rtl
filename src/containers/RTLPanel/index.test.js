import React from 'react'
import {shallow} from 'enzyme'
import RTLPanel from '.'

describe('RTLPanel', () => {
  let api
  let channel
  let wrapper

  beforeEach(() => {
    api = {
      getQueryParam: jest.fn()
    }
    channel = {
      on: jest.fn(),
      removeListener: jest.fn(),
      emit: jest.fn()
    }
  })

  describe('without query parameter', () => {
    beforeEach(() => {
      api.getQueryParam.mockReturnValue(undefined)
      wrapper = shallow(<RTLPanel api={api} channel={channel} />)
    })

    it('renders', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('with query parameter', () => {
    beforeEach(() => {
      api.getQueryParam.mockReturnValue('rtl')
      wrapper = shallow(<RTLPanel api={api} channel={channel} />)
    })

    it('renders', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
