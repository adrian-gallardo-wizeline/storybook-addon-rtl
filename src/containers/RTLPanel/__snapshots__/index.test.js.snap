// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RTLPanel with query parameter renders 1`] = `
<RTLPanel
  checked={true}
  onChange={[Function]}
/>
`;

exports[`RTLPanel without query parameter renders 1`] = `
<RTLPanel
  checked={false}
  onChange={[Function]}
/>
`;
